suit_generator/cmd_sign.py

```diff
@@ -0,0 +1,321 @@
+                    RecursiveSigner(
+                        self._load_dependency(dep),
+                        envelope_json["dependencies"][dep],
+                        dep,
+                        self.sign_script,
+                        self.kms_script,
+                        self.alg,
+                        self.context,
+                    )
+                )
+
+    def _load_dependency(self, dependency_name):
+        """Load the dependency from the envelope."""
+        if dependency_name not in self.envelope.value:
+            raise ValueError(f"Dependency {dependency_name} not found in {self.envelope_name}.")
+        if not isinstance(self.envelope.value[dependency_name], bytes):
+            raise ValueError(f"Dependency {dependency_name} in {self.envelope_name} is invalid.")
+        try:
+            dependency_envelope = cbor2.loads(self.envelope.value[dependency_name])
+        except cbor2.CBORDecodeError:
+            raise ValueError(f"Failed decoding dependency {dependency_name} in {self.envelope_name}")
+        if not isinstance(dependency_envelope, cbor2.CBORTag):
+            raise ValueError(f"Dependency {dependency_name} in {self.envelope_name} is not a valid envelope.")
+
+        return dependency_envelope
+
+    def _sign(self):
+        self.envelope = self.signer.sign_envelope(
+            self.envelope,
+            self.key_name,
+            self.key_id,
+            self.alg,
+            self.context,
+            self.kms_script,
+            self.already_signed_action,
+        )
+
+    def recursive_sign(self):
+        """Recursively sign the envelope and the dependencies."""
+        for dep in self.dependencies:
+            self.envelope.value[dep.envelope_name] = cbor2.dumps(dep.recursive_sign())
+        if not self.omit_signing:
+            self._sign()
+        return self.envelope
+
+
+def auto_int(x):
+    """Convert a string to an integer - allowing for formats like 0x... needed for argparse."""
+    return int(x, 0)
+
+
+def add_arguments(parser):
+    """Add additional arguments to the passed parser."""
+    cmd_sign_arg_parser = parser.add_parser(SIGN_CMD, help="Sign a SUIT envelope.")
+
+    cmd_sign_subparsers = cmd_sign_arg_parser.add_subparsers(
+        dest="sign_subcommand", required=True, help="Choose sign subcommand"
+    )
+
+    cmd_sign_single_level = cmd_sign_subparsers.add_parser(
+        SIGN_SINGLE_LEVEL_CMD,
+        help="Sign a single envelope - one level signing",
+    )
+    cmd_sign_single_level.add_argument("--input-envelope", required=True, type=Path, help="Input envelope.")
+    cmd_sign_single_level.add_argument("--output-envelope", required=True, type=Path, help="Output envelope.")
+    cmd_sign_single_level.add_argument(
+        "--key-name", required=True, type=str, help="Name of the key used by the KMS to identify the key."
+    )
+    cmd_sign_single_level.add_argument(
+        "--key-id",
+        required=True,
+        type=auto_int,
```

Comment:
nit: Instead of additional auto_int function you could use `lambda x: int(x, 0)`